src/components/EditCoursePage/index.jsx

```diff
@@ -0,0 +1,213 @@
+        2. Putting the entitlement into an array and also adding in the sku
+          (required for updating price)
+        3. Renaming the image and video fields to correspond to what course-discovery is expecting
+        4. Setting the uuid so we can create the url to send to course-discovery
+    */
+    const {
+      courseInfo: {
+        data: {
+          uuid,
+          entitlements,
+        },
+      },
+    } = this.props;
+    const courseData = options;
+    courseData.subjects = [
+      courseData.subjectPrimary,
+      courseData.subjectSecondary,
+      courseData.subjectTertiary,
+    ].filter(subject => !!subject);
+    courseData.entitlements = entitlements && [{
+      mode: courseData.mode,
+      price: courseData.price,
+      sku: entitlements[0].sku,
+    }];
+    courseData.image = courseData.imageSrc;
+    courseData.video = courseData.videoSrc;
+    courseData.uuid = uuid;
+    return this.props.editCourse(courseData);
+  }
+
+  render() {
+    if (!this.props.courseInfo || !this.props.courseOptions) {
+      return (
+        <StatusAlert
+          id="error"
+          alertType="danger"
+          title="Course Edit Form failed to load: "
+          message="Course information unavailable."
+        />
+      );
+    }
+
+    const {
+      courseInfo,
+      courseInfo: {
+        data: {
+          title,
+          key,
+          short_description,
+          full_description,
+          outcome,
+          subjects,
+          image,
+          prerequisites_raw,
+          level_type,
+          learner_testimonials,
+          faq,
+          additional_information,
+          syllabus_raw,
+          video,
+          entitlements,
+        },
+      },
+      courseOptions,
+    } = this.props;
+
+    // If we want to keep a lot of the logic in the lower return,
+    // we have to do all this weird '&&' logic to make sure the data is there.
+    const keyParts = key && key.split(/\+|\//);
+    const number = keyParts && keyParts[keyParts.length - 1];
+    const subjectMap = subjects && subjects.map(x => x.slug);
+    const subjectPrimary = subjectMap && subjectMap[0];
+    const subjectSecondary = subjectMap && subjectMap[1];
+    const subjectTertiary = subjectMap && subjectMap[2];
+    const imageSrc = image && image.src;
+    const videoSrc = video && video.src;
+    const entitlement = entitlements && entitlements[0];
+    const mode = (entitlement && entitlement.mode) || 'Audit';
+    const price = entitlement && entitlement.price;
+
+    return (
+      <React.Fragment>
+        <Helmet>
+          <title>{`Course - ${title}`}</title>
+        </Helmet>
+
+        <div className="container-fluid">
+          <div className="row justify-content-md-center my-3 ">
+            <div className="col-6">
+              { (!this.state.startedFetching || courseInfo.isFetching ||
+                courseOptions.isFetching) &&
+                <LoadingSpinner /> }
+              { courseInfo.error && (
```

Comment:
Could you merge these into one status alert with a single conditional?

Maybe something like
```
error = //Combine the error messages
{ error && (
<StatusAlert
   id="error"
   alertType="danger"
   title="Course Edit Form failed to load: "
   message={courseInfo.error}
/>
)}
```
This is just sudo code of course, but it seems possible.  Unless the intent is to show multiple status alerts, which is also valuable.

Reply:
I don't have a preference. Happy to do whichever. I haven't looked into what the error messages would look like so I'm not really sure whether it makes more sense to combine or not.

Reply:
I think it is ok to keep it as is. I looked at it again and it 2 error boxes will be fine.  We can adjust with experience team feedback as needed.